amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/retentionschedule/RetentionScheduleActionRelation.java

```diff
@@ -0,0 +1,225 @@
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.alfresco.module.org_alfresco_module_rm.util.RMParameterCheck.checkNotBlank;
+import static org.alfresco.util.ParameterCheck.mandatory;
+
+/**
+ * Retention schedule action relation is used to perform the retention schedule step operations.
+ */
+@RelationshipResource(name = "retention-steps", entityResource = RetentionScheduleEntityResource.class, title = "Retention Schedule Action")
+@Data
+public class RetentionScheduleActionRelation implements RelationshipResourceAction.Read<RetentionScheduleActionDefinition>,
+        RelationshipResourceAction.Create<RetentionScheduleActionDefinition>
+{
+
+    private FilePlanComponentsApiUtils apiUtils;
+    protected NodeService nodeService;
+    private RecordsManagementServiceRegistry service;
+    private ApiNodesModelFactory nodesModelFactory;
+
+    @Override
+    @WebApiDescription(title="Create a retention schedule step for the particular retention schedule using the 'retentionScheduleId'")
+    public List<RetentionScheduleActionDefinition> create(String retentionScheduleId, List<RetentionScheduleActionDefinition> nodeInfos, Parameters parameters)
+    {
+        checkNotBlank("retentionScheduleId", retentionScheduleId);
+        mandatory("entity", nodeInfos);
+        mandatory("parameters", parameters);
+        NodeRef retentionScheduleNodeRef = apiUtils.lookupAndValidateNodeType(retentionScheduleId, RecordsManagementModel.TYPE_DISPOSITION_SCHEDULE);
+        // validation for the order of the step
+        retentionScheduleStepValidation(retentionScheduleNodeRef, nodeInfos.get(0));
+        // request property validation
+        retentionScheduleRequestValidation(nodeInfos.get(0));
+        // create the parameters for the action definition
+        Map<QName, Serializable> actionDefinitionParams = nodesModelFactory.createRetentionActionDefinitionParams(nodeInfos.get(0));
+        // create the child association from the schedule to the action definition
+        NodeRef actionNodeRef = this.nodeService.createNode(retentionScheduleNodeRef,
+                RecordsManagementModel.ASSOC_DISPOSITION_ACTION_DEFINITIONS,
+                QName.createQName(NamespaceService.CONTENT_MODEL_1_0_URI,
+                        QName.createValidLocalName(nodeInfos.get(0).getName())),
+                RecordsManagementModel.TYPE_DISPOSITION_ACTION_DEFINITION, actionDefinitionParams).getChildRef();
+        DispositionSchedule schedule = new DispositionScheduleImpl(service, nodeService, retentionScheduleNodeRef);
```

Comment:
can we name this variable as dispositionSchedule

Reply:
done.